src/compiler/checker/lines_0_1000.rs

```diff
@@ -160,40 +160,36 @@ pub fn create_type_checker<TTypeCheckerHost: TypeCheckerHost>(
             )
             .into(),
     ));
-    type_checker.any_type = Some(Rc::new(
+    type_checker.any_type = Some(
         type_checker
             .create_intrinsic_type(TypeFlags::Any, "any")
             .into(),
-    ));
-    type_checker.number_type = Some(Rc::new(
+    );
+    type_checker.number_type = Some(
         type_checker
             .create_intrinsic_type(TypeFlags::Number, "number")
             .into(),
-    ));
-    type_checker.bigint_type = Some(Rc::new(
+    );
+    type_checker.bigint_type = Some(
         type_checker
             .create_intrinsic_type(TypeFlags::BigInt, "bigint")
             .into(),
-    ));
-    let true_type: Rc<Type> = Rc::new(
-        FreshableIntrinsicType::new(
-            type_checker.create_intrinsic_type(TypeFlags::BooleanLiteral, "true"),
-        )
-        .into(),
-    );
-    let regular_true_type: Rc<Type> = Rc::new(
-        FreshableIntrinsicType::new(
-            type_checker.create_intrinsic_type(TypeFlags::BooleanLiteral, "true"),
-        )
-        .into(),
     );
+    let true_type: Rc<Type> = FreshableIntrinsicType::new(
+        type_checker.create_intrinsic_type(TypeFlags::BooleanLiteral, "true"),
+    )
+    .into();
+    let regular_true_type: Rc<Type> = FreshableIntrinsicType::new(
+        type_checker.create_intrinsic_type(TypeFlags::BooleanLiteral, "true"),
+    )
+    .into();
     match &*true_type {
         Type::IntrinsicType(intrinsic_type) => match intrinsic_type {
             IntrinsicType::FreshableIntrinsicType(freshable_intrinsic_type) => {
                 freshable_intrinsic_type
                     .regular_type
-                    .init(&regular_true_type, true);
-                freshable_intrinsic_type.fresh_type.init(&true_type, true);
+                    .init(&regular_true_type, false);
```

Comment:
It's now not true that `regular_true_type` is a unique `Rc` reference because we've stored a second reference in `_type_wrapper`